test/cqlpy/test_tablets.py

```diff
@@ -323,3 +323,33 @@ def change_dc_rf(new_rf):
             change_dc_rf(10)  # increase RF by 2+ should fail
         with pytest.raises(InvalidRequest):
             change_dc_rf(0)  # decrease RF by 2+ should fail
+
+
+def test_tablet_hints(cql, skip_without_tablets):
+    def describe_table(cql, table):
+        return cql.execute(f"DESC TABLE {table}").one().create_statement
+
+    ksdef = "WITH REPLICATION = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND TABLETS = {'enabled': false};"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        tablet_hints = "tablet_hints = {'min_tablet_count': '100'}"
+        with new_test_table(cql, keyspace, "pk int PRIMARY KEY, c int", extra=f"WITH {tablet_hints}") as table:
+            assert tablet_hints in describe_table(cql, table)
+            # ALTER TABLE for other options does not affect existing tablet_hints
+            cql.execute(f"ALTER TABLE {table} WITH gc_grace_seconds = 42")
+            assert tablet_hints in describe_table(cql, table)
+            # Resetting tablet_hints to an empty map drops all hints
+            tablet_hints = "tablet_hints = {}"
+            cql.execute(f"ALTER TABLE {table} WITH {tablet_hints}")
+            assert tablet_hints in describe_table(cql, table)
+            # New tablet_hints can be added by ALTER TABLE
+            tablet_hints = "tablet_hints = {'expected_data_size_in_gb': '50', 'min_tablet_count': '100'}"
+            cql.execute(f"ALTER TABLE {table} WITH {tablet_hints}")
+            assert tablet_hints in describe_table(cql, table)
+            # tablet_hints with zero values are dropped
+            tablet_hints = "tablet_hints = {'expected_data_size_in_gb': '0', 'min_tablet_count': '100'}"
+            cql.execute(f"ALTER TABLE {table} WITH {tablet_hints}")
+            assert "tablet_hints = {'min_tablet_count': '100'}" in describe_table(cql, table)
+            # tablet_hints are set as a whole, replacing the previously set hints
+            tablet_hints = "tablet_hints = {'min_per_shard_tablet_count': '3.14'}"
```

Comment:
The per_shard_tablet_count is allowed to be a floating point? Is this intentional?

Reply:
Maybe you can add a comment saying you're intentionally checking that floating numbers work, to reduce surprise in whoever reads this test in the future.

To be honest, I still don't know what  a minimum of 3.14 tablets per shard actually means :-( Would 3 tablets be allowed on some shard, or it has to be higher than 3.14 so needs to be 4 tablets?

Reply:
`min_per_shard_tablet_count` doesn't mean that each shard should have at least that, it means that the average per-shard tablet count is at least that, so it's really `min_avg_per_shard_tablet_count`

Reply:
> `min_per_shard_tablet_count` doesn't mean that each shard should have at least that, it means that the average per-shard tablet count is at least that, so it's really `min_avg_per_shard_tablet_count`

The question then is why the hint needs to be floating point?
Can't the user just provide the std::floor of that average?

Reply:
If min_per_shard_tablet_count isn't the minimum, maybe we shouldn't call it that?
This parameter needs to be something that users will be able to understand (and of course be clearly documented).

Reply:
> > `min_per_shard_tablet_count` doesn't mean that each shard should have at least that, it means that the average per-shard tablet count is at least that, so it's really `min_avg_per_shard_tablet_count`
> 
> The question then is why the hint needs to be floating point? Can't the user just provide the std::floor of that average?

The original reasoning still stands: https://github.com/scylladb/scylladb/pull/22090#discussion_r1901760324

Reply:
> If min_per_shard_tablet_count isn't the minimum, maybe we shouldn't call it that? This parameter needs to be something that users will be able to understand (and of course be clearly documented).

It is a minimum, just for the average. The count and average count per shared can still be higher due than this parameter tells due to other factors, like table size.

Reply:
> The original reasoning still stands: https://github.com/scylladb/scylladb/pull/22090#discussion_r1901760324

Can you please provide an example demonstrating why it is needed and giving an integer (floor) wouldn't work?

Reply:
tablet_count = min_per_shard_tablet_count * shard_count

shard_count = 100

min_per_shard_tablet_count = 1
tablet_count = 100 (128, nearest pow2)

min_per_shard_tablet_count = 2
tablet_count = 200 (256)

Impossible to achieve tablet count of 64.

With fractions:

min_per_shard_tablet_count = 0.64
tablet_count = 0.64 * 100 = 64

Reply:
ok, I get your point, though one could use min_tablet count for that too.